dotcom-rendering/src/web/components/Braze.tsx

```diff
@@ -0,0 +1,150 @@
+import { useState, useEffect } from 'react';
+import { ArticleDesign, storage } from '@guardian/libs';
+import { BrazeMessagesInterface } from '@guardian/braze-components';
+import {
+	incrementWeeklyArticleCount,
+	getWeeklyArticleHistory,
+} from '@guardian/support-dotcom-components';
+import { WeeklyArticleHistory } from '@guardian/support-dotcom-components/dist/dotcom/src/types';
+import { parse } from '@root/src/lib/slot-machine-flags';
+import { buildBrazeMessages } from '../lib/braze/buildBrazeMessages';
+import { SlotBodyEnd } from './SlotBodyEnd/SlotBodyEnd';
+import { StickyBottomBanner } from './StickyBottomBanner/StickyBottomBanner';
+import { hasOptedOutOfArticleCount } from '../lib/contributions';
+import { incrementDailyArticleCount } from '../lib/dailyArticleCount';
+import { useOnce } from '../lib/useOnce';
+
+type Props = {
+	idApiUrl: string;
+	contentType: string;
+	sectionName?: string;
+	shouldHideReaderRevenue: boolean;
+	isMinuteArticle: boolean;
+	isPaidContent: boolean;
+	tags: TagType[];
+	contributionsServiceUrl: string;
+	stage: string;
+	section: string;
+	isPreview: boolean;
+	isSensitive: boolean;
+	switches: Switches;
+	keywordIds: string;
+	pageId: string;
+	slotMachineFlags?: string;
+	format: ArticleFormat;
+};
+
+function shouldShowSlotBodyEnd(
+	format: ArticleFormat,
+	switches: Switches,
+	slotMachineFlags?: string,
+) {
+	if (!switches.slotBodyEnd) return false;
+	if (!parse(slotMachineFlags || '').showBodyEnd) return false;
```

Comment:
Could somebody please check that this code is the same as:

```
	const showBodyEndSlot =
		parse(CAPI.slotMachineFlags || '').showBodyEnd ||
		CAPI.config.switches.slotBodyEnd;
```

This `parse` function is hard to reason about

Reply:
The logic looks the same in each case. Spliting it out as above is nice and clear.
A tiny `slot-machine-flags.test` could be useful. 

What is the source of `CAPI.slotMachineFlags` & `CAPI.config.switches`?

Reply:
> What is the source of CAPI.slotMachineFlags & CAPI.config.switches?

I'm honestly not sure. I think switches maybe come from a tool around Composer?  `slotMachineFlags` I know nothing about @sndrs any ideas?